test/server/cards/05_LOF/units/MythosaurFolkloreAwakened.spec.ts

```diff
@@ -0,0 +1,256 @@
+                expect(context.mythosaur.exhausted).toBeTrue();
+                expect(context.player2.discard.length).toBe(5);
+            });
+
+            /**
+             *********** Return to Hand Prevention Tests ************
+             **/
+
+            it('prevents friendly upgraded units from being returned to hand by enemy card triggered abilities', function() {
+                const { context } = contextRef;
+
+                // Player 1 uses Kanan Jarrus to give a Shield token to Mythosaur
+                context.player1.clickCard(context.kananJarrus);
+                context.player1.clickPrompt('Give a Shield token to a Creature or Spectre unit.');
+                context.player1.clickCard(context.mythosaur);
+
+                expect(context.mythosaur).toHaveExactUpgradeNames(['shield']);
+
+                // Player 2 plays Cantina Bouncer to return Mythosaur to hand
+                context.player2.clickCard(context.cantinaBouncer);
+                context.player2.clickCard(context.mythosaur);
+
+                // Mythosaur is not returned to hand due to its ability
+                expect(context.mythosaur).toBeInZone('groundArena');
+                expect(context.getChatLogs(1)).toContain('player1 uses Mythosaur to prevent Mythosaur from returning to hand');
+            });
+
+            it('prevents friendly upgraded units from being returned to hand by enemy event abilities', function() {
+                const { context } = contextRef;
+                context.player1.passAction();
+
+                // Player 2 plays Evacuate to return all units to hand
+                context.player2.clickCard(context.evacuate);
+
+                // Satine Kryze is upgraded and should not be returned to hand
+                expect(context.satineKryze).toBeInZone('groundArena');
+
+                // All other units should be returned to hand
+                expect(context.mythosaur).toBeInZone('hand', context.player1);
+                expect(context.razorCrest).toBeInZone('hand', context.player1);
+                expect(context.grogu).toBeInZone('hand', context.player2);
+                expect(context.getChatLogs(3)).toContain('player1 uses Satine Kryze\'s gained ability from Mythosaur to prevent Satine Kryze from returning to hand');
+            });
+
+            it('does not prevent friendly upgraded units from being returned to hand by friendly card abilities', function() {
+                const { context } = contextRef;
+
+                // Player 1 plays Waylay to return Satine Kryze to hand
+                context.player1.clickCard(context.waylay);
+                context.player1.clickCard(context.satineKryze);
+
+                // Satine Kryze is returned to hand
+                expect(context.satineKryze).toBeInZone('hand', context.player1);
+            });
+        });
+
+        describe('Mythosaur\'s second constant ability', function() {
```

Comment:
I would also add one test to check that pilot leaders that are piloting get the mandalorian trait too. I don't think it matters for any card at the moment but who knows what they will release in the future!